Flaky-test: BacklogQuotaManagerTest.testProducerExceptionAndThenUnblockSizeQuota

BacklogQuotaManagerTest.testProducerExceptionAndThenUnblockSizeQuota is flaky. It fails sporadically.

[example failure](https://github.com/apache/pulsar/runs/5134450317?check_suite_focus=true#step:9:585)

```
java.lang.AssertionError: unable to publish due to org.apache.pulsar.client.api.PulsarClientException$TimeoutException: The producer usc-0-0 can not send message to the topic persistent://prop/quotahold/exceptandunblock within given timeout : createdAt 2.002 ns ago, firstSentAt 1618.939 ns ago, lastSentAt 1618.939 ns ago, retryCount 0 expected [false] but found [true]
	at org.testng.Assert.fail(Assert.java:99)
	at org.testng.Assert.failNotEquals(Assert.java:1037)
	at org.testng.Assert.assertFalse(Assert.java:67)
	at org.apache.pulsar.broker.service.BacklogQuotaManagerTest.testProducerExceptionAndThenUnblockSizeQuota(BacklogQuotaManagerTest.java:1135)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/service/BacklogQuotaManagerTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/service/BacklogQuotaManagerTest.java
@@ -1122,6 +1122,8 @@ public class BacklogQuotaManagerTest {
         }
         Thread.sleep((TIME_TO_CHECK_BACKLOG_QUOTA + 1) * 1000);
         // publish should work now
+        producer.close();
+        producer = createProducer(client, topic1);
         Exception sendException = null;
         gotException = false;
         try {

